app/blueprints/assessments/activity_trail.py

```diff
@@ -0,0 +1,307 @@
+import inspect
+from dataclasses import asdict
+from dataclasses import dataclass
+from dataclasses import field
+from datetime import datetime
+
+from app.blueprints.services.data_services import get_bulk_accounts_dict
+from app.blueprints.services.models.assessor_task_list import AssessorTaskList
+from app.blueprints.services.models.flag import FlagType
+from app.blueprints.tagging.models.tag import AssociatedTag
+from flask import current_app
+from flask_wtf import FlaskForm
+from wtforms import BooleanField
+from wtforms import StringField
+
+
+@dataclass
+class BaseModel:
+    @classmethod
+    def from_list(cls, data_list: list[dict]):
+        return [
+            cls(
+                **{
+                    k: v
+                    for k, v in d.items()
+                    if k in inspect.signature(cls).parameters
+                    and k != "date_created"
+                },
+                date_created=cls._format_date(d.get("date_created")),
+            )
+            for d in data_list
+        ]
+
+    @staticmethod
+    def _format_date(date_str):
+        if date_str:
+            try:
+                current_app.logger.info("Formatting date")
+                return datetime.fromisoformat(date_str).strftime(
+                    "%Y-%m-%d %H:%M:%S"
+                )
+            except ValueError:
+                return date_str
+        return date_str
```

Comment:
We prob don't need to log that we're formatting the date, could remove that

Also, you could supress the error since the end return the date string at the end anyways:

```suggestion
    @staticmethod
    def _format_date(date_str):
        if date_str:
            with contextlib.suppress(ValueError)
                return datetime.fromisoformat(date_str).strftime(
                    "%Y-%m-%d %H:%M:%S"
                )
        return date_str
```

Reply:
Thank you. Sure!